File: ruby/samples/get_all_municipalities.rb
# Load the gem
require 'intrinio-sdk'

# Setup authorization
Intrinio.configure do |config|
  config.api_key['api_key'] = 'YOUR_API_KEY'
end

municipality_api = Intrinio::MunicipalityApi.new

opts = { 
  next_page: nil # String | Gets the next page of data from a previous API call
}

begin
  result = municipality_api.get_all_municipalities(opts)
  municipalities = result.municipalities

  puts "#{municipalities.size} municipalities found!"

  municipalities.each.with_index(1) do |municipality, index|
    puts ""
    puts "#{index}.  #{municipality.government_name}"
    puts "-------------------------------------------------------------------------"
    puts "ID:              #{municipality.id}"
    puts "Census ID:       #{municipality.census_id}"
    puts "Government type: #{municipality.government_type}"
    puts "Area type:       #{municipality.area_type}"
    puts "Area name:       #{municipality.area_name}"
    puts "Location:        #{municipality.city}, #{municipality.state} #{municipality.zip}"
  end
rescue Intrinio::ApiError => e
  puts "Exception when calling MunicipalityApi->get_all_municipalities: #{e}"
end

File: javascript-web/samples/getAllMunicipalities.js
var intrinioSDK = require('intrinio-sdk');
intrinioSDK.ApiClient.instance.authentications['ApiKeyAuth'].apiKey = "YOUR API KEY";

var municipalityAPI = new intrinioSDK.MunicipalityApi();

var opts = { 
  'nextPage': null // String | Gets the next page of data from a previous API call
};

municipalityAPI.getAllMunicipalities(opts).then(function(data) {
  $('#apiResult').html(JSON.stringify(data, undefined, 2));

  var municipalities = data.municipalities;

  console.log(municipalities.length + " municipalities found!")

  municipalities.forEach(function(municipality, index) {
    console.log();
    console.log((index + 1) + ".  " + municipality.government_name);
    console.log("-------------------------------------------------------------------------");
    console.log("ID:              " + municipality.id);
    console.log("Census ID:       " + municipality.census_id);
    console.log("Government type: " + municipality.government_type);
    console.log("Area type:       " + municipality.area_type);
    console.log("Area name:       " + municipality.area_name);
    console.log("Location:        " + municipality.city + " " + municipality.state + " " + municipality.zip);
  });
}, function(error) {
  console.error(error);
});
